samples/cpp/text_generation/structured_output_generation.cpp

```diff
@@ -0,0 +1,70 @@
+// Copyright (C) 2023-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "openvino/genai/llm_pipeline.hpp"
+
+int main(int argc, char* argv[]) try {
+    if (argc < 2 || argc > 3) {
+        throw std::runtime_error(std::string{"Usage: "} + argv[0] + " <MODEL_DIR> <DEVICE>");
+    }
+    std::string prompt;
+    std::string models_path = argv[1];
+
+    // Default device is CPU; can be overridden by the second argument
+    std::string device = (argc == 3) ? argv[2] : "CPU";  // GPU, NPU can be used as well
+    ov::genai::LLMPipeline pipe(models_path, device);
+    
+    ov::genai::GenerationConfig config;
+    config.max_new_tokens = 1000;
+    config.do_sample = false;
+
+    std::string json_chema = R"({"$defs": {"Step": {"properties": {"explanation": {"title": "Explanation", "type": "string"}, "output": {"title": "Output", "type": "string"}}, "required": ["explanation", "output"], "title": "Step", "type": "object"}}, "properties": {"steps": {"items": {"$ref": "#/$defs/Step"}, "title": "Steps", "type": "array"}, "final_answer": {"title": "Final Answer", "type": "string"}}, "required": ["steps", "final_answer"], "title": "MathReasoning", "type": "object"})";
```

Comment:
Does it make sense to make schema more readable? Like format it as a multiline string with indentations?

Reply:
done